calico_kubernetes/calico_kubernetes.py

```diff
@@ -97,6 +147,223 @@ def _configure_interface(self):
         print('Finished configuring network interface')
         return ep
 
+    def _configure_profile(self, endpoint):
+        """
+        Configure the calico profile for a pod.
+
+        Currently assumes one pod with each name.
+        """
+        print('Configuring Pod Profile')
+        profile_name = self.pod_name
+
+        if self._datastore_client.profile_exists(profile_name):
+            print("A profile with name %s already exists" % profile_name)
+            sys.exit(1)
+        else:
+            self._datastore_client.create_profile(profile_name)
+
+        pod = self._get_pod_config()
+
+        self._apply_rules(profile_name)
+
+        self._apply_tags(profile_name, pod)
+
+        # Also set the profile for the workload.
+        print('Setting profile %s on endpoint %s' %
+              (profile_name, endpoint.endpoint_id))
+        self._datastore_client.set_profiles_on_endpoint(
+            profile_name, endpoint_id=endpoint.endpoint_id
+        )
+        print('Finished configuring profile.')
+
+    def _container_add(self, container_name, ip, interface):
+        """
+        Add a container (on this host) to Calico networking with the given IP.
+
+        :param container_name: The name or ID of the container.
+        :param ip: An IPAddress object with the desired IP to assign.
+        :param interface: The name of the interface in the container.
+        """
+        # Obtain information from Docker Client and validate container state
+        info = self._get_container_info(container_name)
+        self._validate_container_add(info)
+
+        # Obtain ip pool from datastore client, validate pool, and assign
+        # ip address through IPAM Client
+        ip = IPAddress(ip)
+        self._assign_address_in_pool_IPAM(ip)
+
+        # Obtain next hops from datastore client, validate next hop, and
+        # create Endpoint object
+        container_id = info["Id"]
+        ep = self._endpoint_init(container_id, ip)
+
+        # Create the veth, move into the container namespace, add the IP and
+        # set up the default routes.
+        # Update Endpoint in felix
+        pid = info["State"]["Pid"]
+        ep = self._endpoint_complete(ep, pid, ip, interface)
+
+        # Let the caller know what endpoint was created.
+        return ep
+
+    def _container_remove(self, container_name):
+        """
+        Remove the indicated container on this host from Calico networking
+
+        :param container_name: THe name or partial id of the container
+        """
+        # Resolve the name to ID.
+        workload_id = self._get_container_id(container_name)
```

Comment:
Don't call _get_container_id(...) inside `container_add` or `container_remove`.  This relies on the docker_client, and we want to make sure that these APIs are not reliant on docker.